docs_src/design/adr/core/0019-EdgeX-CLI-V2.md

```diff
@@ -0,0 +1,113 @@
+
+3. Take full advantage of the features of the underlying command-line library, [Cobra](https://github.com/spf13/cobra), such as tab-completion of commands.
+
+4. Change the client binary name from `edgex-cli` to `edgex`. The reason is that the "-cli" is superfluous. There is no `edgex` tool and best practice is to choose a simple, memorable word. CLI tool names do not generally include `"-cli"` in the name. (Examples: `go`,`snap`,`kubectl`,`git`,`docker` etc)
+
+5. Add support for the four common endpoints - `/ping`, `/metrics`, `/version`, `/status`
+
+6. Change the syntax of the CLI to require a positional service arg which precedes the command, thus allowing overlap of command names across services. The service name can either be an abbreviation or the full name of the service . Examples:
+
+    - `edgex data ping`
+    - `edgex d ping`
+    - `edgex metadata version`
+    - `edgex m version`
+    - `edgex metadata device list`
+    - `edgex command device list`
+
+    An alternative solution is to use a positional argument only for the common endpoints but specify it after the command or to have a regular arg like `-s`/`--service` `<name>`).
+
+
+7. Implement all required V2 endpoints for core services
+
+    **Core Command** (`edgex (c)ommand`)
+    - **`command`** `get | put | list` *(we might want to replace get/put with read/write to be more consistent w/other commands)*
+
+    **Core Data** (`edgex (d)ata`)
+    - **`event`** `add | count | list | rm | scrub**`
+    - **`reading`** `count | list`
+
+    **Metadata** (`edgex (m)etadata`)
+    
+    - **`device`**  `add | adminstate | list | operstate | rm | update`
+    - **`deviceprofile`**  `add | list | rm | update`
+    - **`deviceservice`** ` add | list | rm | update`
+    - **`provisionwatcher`**  `add | list | rm | update`
+    
+    **Support Notifications** ( `edgex (n)otifications`)
+    - **`notification`** `add | list | rm`
+    - **`subscription`** `add | list | rm`
+
+   **Support Scheduler** ( `edgex (s)cheduler`)
+    - **`interval`** `add | list | rm | update` 
+
+
+    **Common endpoints in all services**
+    - **`version`**
+    - **`ping`**
+    - **`metrics`**
+    - **`status`** 
+
+
+
+8. Use the [Viper](https://github.com/spf13/viper) project for configuration file support, instead of BurntSushi. It's from the same people as the Cobra library and they work well together.
+
+9.  Currently, some commands default to always displaying GUIDs in objects when they're not really needed. Change this so that by default GUIDs aren't displayed, but add a flag which causes them to be displayed. 
+
+10. **scrub** may not work with Redis being secured by default. That might also apply to the top-level `db` command (used to wipe the entire db). If so, then the `db` command will be removed.
```

Comment:
Are we assuming this V2 CLI is going to work with security on?  Existing CLI does not.  I am ok if first release of V2 only works with security off.  For the most part, this is a developers tool.  Or make this a stretch goal.

From a developers standpoint, the scrub command is important.  Where a command is not capable to run because of security, we should still offer it for non-secure mode.

Reply:
Yes - as a stretch goal. I updated this to say "*scrub** may not work with Redis being secured by default. That might also apply to the top-level `db` command (used to wipe the entire db). If so, then the commands will be disabled in secure mode, but permitted in non-secure mode. "